torch/export/__init__.py

```diff
@@ -934,34 +936,75 @@ def export(
-    with the provided `args` and `kwargs`. It records the PyTorch operations
-    invoked during execution to produce the ExportedProgram.
+        args: Example positional inputs.
 
+        kwargs: Optional example keyword inputs.
 
-    **Acceptable input/output types**
+        constraints: An optional list of constraints on the dynamic arguments
+         that specify their possible range of shapes. By default, shapes of
+         input torch.Tensors are assumed to be static. If an input torch.Tensor
+         is expected to have dynamic shapes, please use :func:`dynamic_dim`
+         to define `Constraint` objects that specify the dynamics and the possible
+         range of shapes. See :func:`dynamic_dim` docstring for examples on
+         how to use it.
 
+    Returns:
+        An :class:`ExportedProgram` containing the traced callable.
+
+    **How it works**
+    :func:`export` operates by using TorchDynamo to symbolically trace the
+    bytecode of the passed in Python callable, inlining functions and control
+    flow and recording the Tensor calls into an FX graph.  Export then
+    applies further lowering passes to normalize these calls into a
+    functionalized core ATen operator set.
+
+    Compared to :func:`torch.fx.symbolic_trace`, :func:`export` operates at
+    the bytecode level, giving it the ability to trace arbitrary Python constructs
+    not limited by what Python operator overloading supports.  We also keep
+    fine-grained track of Tensor metadata, so that conditionals on things like
+    `tensor.size(i)` do not fail tracing, and then normalize the graph to ATen operators.
+    In general, you should expect :func:`export` to work on more user programs, but to
+    produce lower-level graphs.  You can still use :func:`torch.fx.symbolic_trace` as
+    a pre-processing step before :func:`export`
+
+    Compared to :func:`torch.jit.script`, :func:`export` does not capture Python
+    control flow or data structures.  We support more Python language features than
+    TorchScript (as it is easier to have comprehensive coverage over Python
+    bytecodes).  The resulting graphs are simpler and only have straight line
+    control flow (except for explicit control flow operators).
+
+    Compared to :func:`torch.jit.trace`, :func:`export` is sound: it is able to
+    trace code that performs integer computation on sizes and records all of the
+    side-conditions necessary to show that a particular trace is valid for other inputs.
+
+    **Acceptable input/output types**
     Acceptable types of inputs (for `args` and `kwargs`) and outputs include:
 
     - Primitive types, i.e. `torch.Tensor`, `int`, `float`, `bool` and `str`.
-    - Dataclasses (must be registered with torch._export.utils.register_dataclass_as_pytree_node` first)
     - (Nested) Data structures comprising of `dict`, `list`, `tuple`, `namedtuple` and
-      `OrderedDict` containing all above types.
+      `OrderedDict` containing primitive types.
```

Comment:
Oh no, why?

Reply:
Nested data structure is supported, I guess my change made it ambiguous, reverted.